apis/database.js

```diff
@@ -80,9 +112,35 @@ export async function addLocalLocation(locationData) {
 
   const createdAt = moment.utc(locationObject.created).toISOString();
   await executeTransaction(
-    'insert into locations (key, coordinateKey, createdAt, resources, status, uploaded) values (?, ?, ?, ?, ?, ?)',
-    [key, key, createdAt, resourcesString, status, 0]
+    'insert into locations (key, coordinateKey, createdAt, resources, status, uploaded, updated) values (?, ?, ?, ?, ?, ?, ?)',
+    [key, key, createdAt, resourcesString, status, 0, 0]
   );
 
   return locationObject;
 }
+
+export async function createOrUpdateDatabase() {
+  const version = await SecureStore.getItemAsync(DATABASE_VERSION_KEY);
+
+  await createDatabases();
+
+  if (version === Constants.manifest.extra.localDatabaseVersion) {
+    return Promise.resolve();
+  }
+
+  const locations = await fetchLocalLocations();
+
+  await clearDatabase();
```

Comment:
This feels extremely dangerous. If something went wrong we would lose all the data ... I wonder if we could do something side by side.

Reply:
I suppose we could create a temporary database and store all the locations we pulled out of the old one and then delete it after we have populated the new one again. How would you feel about that?